src/NATS.Net.OpenTelemetry/OpenTelemetryTracingExtensions.cs

```diff
@@ -0,0 +1,23 @@
+using NATS.Client.Core;
+
+// ReSharper disable once CheckNamespace - Place in OTEL namespace for discoverability
+namespace OpenTelemetry.Trace;
+
+public static class OpenTelemetryTracingExtensions
+{
+    /// <summary>
+    /// Enables trace collection on activity sources from the NATS.Client nuget package.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+    /// <param name="includeInternal">Include traces from internal messaging used for control flow and lower level usage during high level abstractions (e.g. JetStream, KvStore, etc)</param>
+    /// <returns>The same instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddNatsInstrumentation(this TracerProviderBuilder builder, bool includeInternal = false)
```

Comment:
Is this library needed? It's wrapper around calling a method with these 2 strings.

Reply:
This is pretty standard. Providing a second package that provides a strongly typed extension to add the instrumentation.

What we've found is that people don't want the magic strings for `AddSource()` and they want the semantic meaning of "Adding instrumentation". It's also providing the semantics and documentation around what internal vs standard telemetry is.

Adding a second package means that you don't need pollute the main packages with OpenTelemetry. That said, that will be required for baggage to work properly at some point soon though.

Reply:
I get why but IMO, I wouldn't use this package because it does so little.